### PR TITLE
Sync puzzle on game start

### PR DESCRIPTION
**Pull request by wontheone1**
synchronize the puzzle state between clients on game start

A lot of refactorings...
This was tricky to do it and at first I was stuck because of circular dependency,
finally I am finishing this feature.

To start puzzle with existing state that's already stored in server,
web-socket needs to get the state after handshake, and then it should start the game.
So web-socket ns needs game ns to call `start-game!` function. However,
game ns needs web-socket to send the sprites state. This mutual dependency is not allowed in Clojure. But putting everything in util or in one ns is ofc bad practice.
I solved this by passing `send-sprites-state!` function as an argument to when web-sck ns calls `game-start!`. This can be a general solution when encounters circular dependency.

In summary, when encounters circular dependency require one from another and pass needed stuff as an argument.

Once I figured this out, the rest was breeze.

### CODE REVIEW

**Comment by flyjwayur** [comment]
I was also stuck with a same problem(circular dependency), when I tried to make the time/flip record. I I felt how much you consider all the possibilities to solve the problem.  It was the best solution ever! I learned a lot from your example.  :D 👍
